pkg/ring/tokens.go

```diff
@@ -0,0 +1,83 @@
+package ring
+
+import (
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"os"
+	"path"
+)
+
+const (
+	// TokenVersion1 is the version is a simple list of tokens.
+	TokenVersion1 = 1
+
+	// Name of the file in which tokens is stored.
+	tokensFileName = "tokens"
```

Comment:
Can we make it take a full path rather than have a directory and then hardcode the file name?

Reply:
@pracucci had the same suggestion. [this](https://github.com/cortexproject/cortex/pull/1750#discussion_r341284749) was my comment for it. Does it make sense?